### PR TITLE
Fix precision of cpu to millicore and memory to Mi

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes https://github.com/openshift/origin/issues/8391

Fixes code to round down cpu to nearest millicore with a floor of 1m.
Fixes code to round down memory to nearest Mi with a floor of 1Mi.

@spadgett @jwforres @sspeiche @sosiouxme - PTAL

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
I think this should be p1 fix, but I cannot label.

**Comment by derekwaynecarr** [comment]
[test]

**Comment by derekwaynecarr** [comment]
Rationale for p1 fix is I will most likely fix the upstream code to fail validation if it gets fractional bytes for memory or sub-millicore precision on CPU

**Comment by derekwaynecarr** [comment]
@abhgupta - fyi

**Comment by sosiouxme** [comment]
LGTM.

Note for the UI team this rounds memory to bytes; output will presumably still show full byte precision if it doesn't fall on binary values (1023 bytes won't round to 1Ki).

**Comment by jwforres** [comment]
we discussed this with @danmcp and this is going to be updated to round to
nearest whole MiB for memory

**Comment by derekwaynecarr** [comment]
@jwforres @danmcp @sosiouxme PTAL

Fixes code to round down cpu to nearest millicore with a floor of 1m.
Fixes code to round down memory to nearest Mi with a floor of 1Mi.

**Comment by derekwaynecarr** [comment]
@ncdc - this is a blocker bug fix, can you please review?

File: pkg/quota/admission/clusterresourceoverride/admission.go [hunk 1]
```diff
@@ -153,24 +156,37 @@ func (a *clusterResourceOverridePlugin) Admit(attr admission.Attributes) error {
...
 			}
+			resources.Requests[kapi.ResourceMemory] = value
 		}
 		if memFound && a.config.limitCPUToMemoryRatio.Cmp(zeroDec) != 0 {
-			resources.Limits[kapi.ResourceCPU] = resource.Quantity{
-				// float math is necessary here as there is no way to create an inf.Dec to represent cpuBaseScaleFactor < 0.001
-				Amount: multiply(inf.NewDec(int64(float64(memLimit.Value())*cpuBaseScaleFactor), 3), a.config.limitCPUToMemoryRatio),
-				Format: resource.DecimalSI,
+			// float math is necessary here as there is no way to create an inf.Dec to represent cpuBaseScaleFactor < 0.001
+			// cpu is measured in millicores, so we need to scale and round up the value to nearest millicore scale
```
Comment (ncdc) [inline pkg/quota/admission/clusterresourceoverride/admission.go]: comment says "round up" but you use RoundDown below?

Comment (derekwaynecarr) [inline pkg/quota/admission/clusterresourceoverride/admission.go]: will fix comment, round down is correct, comment was missed in my update.

**Comment by derekwaynecarr** [comment]
Comments updated.

**Comment by ncdc** [comment]
Math is fun :smile: 

LGTM

**Comment by derekwaynecarr** [comment]
@smarterclayton - can I have approval and merge?

**Comment by eparis** [comment]
I approve and [merge]!

**Comment by eparis** [comment]
blast you AWS! [merge] already

**Comment by eparis** [comment]
I am relentlessly [merge]ing!

**Comment by eparis** [comment]
flake flake flake [merge] !!!

**Pull request merged**